src/pages/Organization/OrganizationUsers.tsx

```diff
@@ -77,96 +57,129 @@ export default function OrganizationUsers({ id, navOrganizationId }: Props) {
-          {users?.results?.length === 0 ? (
-            <Card className="col-span-full">
-              <CardContent className="p-6 text-center text-gray-500">
-                {t("no_users_found")}
+        <div className="flex gap-2">
+          <Input
+            type="text"
+            placeholder="Search users..."
+            value={qParams.search || ""}
+            onChange={(e) =>
+              updateQuery({
+                search: e.target.value as string,
+              })
+            }
+            className="max-w-sm"
+            data-cy="search-user"
+          />
+        </div>
+        {isLoadingUsers ? (
+          Array.from({ length: 3 }).map((_, i) => (
+            <Card key={i}>
+              <CardContent className="p-6">
+                <div className="flex items-center space-x-4">
+                  <div className="h-12 w-12 rounded-full bg-gray-200 animate-pulse" />
+                  <div className="space-y-2 flex-1">
+                    <div className="h-4 bg-gray-200 rounded animate-pulse w-1/4" />
+                    <div className="h-4 bg-gray-200 rounded animate-pulse w-1/3" />
+                  </div>
+                </div>
               </CardContent>
             </Card>
-          ) : (
-            users?.results?.map((userRole: OrganizationUserRole) => (
-              <Card key={userRole.id} className="h-full">
-                <CardContent className="p-4 sm:p-6">
-                  <div className="flex flex-col h-full gap-4">
-                    <div className="flex gap-4">
-                      <Avatar
-                        name={`${userRole.user.first_name} ${userRole.user.last_name}`}
-                        imageUrl={userRole.user.profile_picture_url}
-                        className="h-12 w-12 sm:h-16 sm:w-16 text-xl sm:text-2xl flex-shrink-0"
-                      />
-                      <div className="flex flex-col min-w-0 flex-1">
-                        <div className="flex flex-col gap-1">
-                          <h1 className="text-base font-bold break-words pr-2">
-                            {userRole.user.first_name} {userRole.user.last_name}
-                          </h1>
-                          <div className="flex items-center gap-2 flex-wrap">
-                            <span className="text-sm text-gray-500 truncate">
-                              {userRole.user.username}
-                            </span>
-                            <UserStatusIndicator user={userRole.user} />
+          ))
+        ) : (
+          <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-4">
+            {users?.results?.length === 0 ? (
+              <Card className="col-span-full">
+                <CardContent className="p-6 text-center text-gray-500">
+                  {t("no_users_found")}
+                </CardContent>
+              </Card>
+            ) : (
+              users?.results?.map((userRole: OrganizationUserRole) => (
```

Comment:
is the typecasting necessary?

```suggestion
              users?.results?.map((userRole) => (
```

Reply:
it's not 👍